packages/gatsby-theme-carbon/src/components/Code/Code.js

```diff
@@ -14,9 +14,19 @@ import Sidebar from './Sidebar';
 
 import useMetadata from '../../util/hooks/useMetadata';
 
-const Code = ({ children, className: classNameProp = '', path, src }) => {
-  const [hasMoreThanNineLines, setHasMoreThanNineLines] = useState(false);
-  const [shouldShowMore, setShouldShowMore] = useState(false);
+const Code = ({
+  children,
+  className: classNameProp = '',
+  overflow: overflowProp = 9,
+  path,
+  src,
+}) => {
+  const overflow = Number(overflowProp); // MDX parses this as a string
+  const isOverFlowEnabled = overflow > 0;
+  const shouldOverflow = children.split('\n').length > overflow;
+  const [isOverflowExpanded, setOverflowExpanded] = useState(
+    !isOverFlowEnabled
```

Comment:
Why does this default to `!isOverFlowEnabled`?

Reply:
The thought was "If overflow is disabled, then, by default, 'expand the overflow'" 

and 

```
isOverflowEnabled ? true : false
```

simplifies to `!isOverFlowEnabled`

_However_ after forgetting about this change 😅 and myself not being able to figure that out in 30 seconds, I rewrote this to move that check into the `getLines` function instead which is probably more clear.